python/tests/hypothesis/arcticdb/test_resample.py

```diff
@@ -0,0 +1,109 @@
+import pandas as pd
+import numpy as np
+import pytest
+from hypothesis import given, settings, assume
+import hypothesis.extra.pandas as hs_pd
+import hypothesis.extra.numpy as hs_np
+import hypothesis.strategies as st
+from arcticdb.util.hypothesis import use_of_function_scoped_fixtures_in_hypothesis_checked
+from arcticdb import QueryBuilder
+from arcticdb.util.test import assert_frame_equal, generic_resample_test
+from arcticdb.util._versions import IS_PANDAS_TWO
+
+
+COLUMN_DTYPE = ["float", "int", "uint"]
+ALL_AGGREGATIONS = ["sum", "mean", "min", "max", "first", "last", "count"]
+MIN_DATE = np.datetime64('1969-01-01')
+MAX_DATE = np.datetime64('2000-01-01')
+
+@st.composite
+def date(draw, min_date, max_date):
+    # Bound the start and end date so that we don't end up with too many buckets eating all RAM
+    # Use some pre-epoch dates.
+    # hs_np.from_dtype's min_value and max_value do not work with dates
+    res = draw(hs_np.from_dtype(np.dtype("datetime64[ns]")))
+    assume(min_date <= res and res <= max_date)
```

Comment:
Prefer using the min and max value args to `from_dtype` for performance reasons

Reply:
My initial attempt was with min and max value but as stated in the comment above it didn't work. It kept returning dates outside of the min/max range.